### ISSUE TITLE
Game crashing functions

### ISSUE THREAD

**Issue by Ketho** [Stanzilla/WoWUIBugs#10]
A few functions crash the game when called without any arguments. This was reported to and acknowledged by TSquare on May 7th.

- Reproduced on: Live 8.1.5 (29981), PTR 8.2.0 (30430)

#### Example
```lua
function CrashMyShit()
	-- ERROR #0 (0x85100000) Assertion failure!
	-- File: d:\buildserver\wow\3\work\shared-checkout\branches\wow-patch-8_2_0-branch\storm\h\sstr.inl
	-- Line: 138
	-- Expr: source
	SetGuildRecruitmentComment()
	
	-- ERROR #0 (0x85100000) Assertion failure!
	-- File: d:\buildserver\wow\3\work\shared-checkout\branches\wow-patch-8_2_0-branch\storm\h\sstr.inl
	-- Line: 235
	-- Expr: string
	GetLFGTypes()
	 
	-- [string "CreateFrame("GameTooltip"):IsUnit()"]
	-- line 1: Usage: d:\buildserver\wow\3\work\shared-checkout\branches\wow-patch-8_2_0-branch\engine\source\lua-5.1\src\lapi.c:IsUnit("unit")
	CreateFrame("GameTooltip"):IsUnit()
	
	-- ERROR #0 (0x85100000) Assertion failure!
	-- File: d:\buildserver\wow\3\work\shared-checkout\branches\wow-patch-8_2_0-branch\engine\source\services\texture.cpp
	-- Line: 2714
	-- Expr: fileName
	CreateFrame("ArchaeologyDigSiteFrame"):SetBorderTexture()
	CreateFrame("ArchaeologyDigSiteFrame"):SetFillTexture()
	CreateFrame("QuestPOIFrame"):SetBorderTexture()
	CreateFrame("QuestPOIFrame"):SetFillTexture()
	CreateFrame("ScenarioPOIFrame"):SetBorderTexture()
	CreateFrame("ScenarioPOIFrame"):SetFillTexture()
end
```

**Comment by Meorawr** [comment]
```lua
	CreateFrame("ArchaeologyDigSiteFrame"):SetBorderTexture()
	CreateFrame("QuestPOIFrame"):SetBorderTexture()
	CreateFrame("ScenarioPOIFrame"):SetBorderTexture()
```

These functions no longer crash the client for me as of build 8.2.0.30827. The `SetFillTexture` ones still do.

Additionally, `CreateFrame("GameTooltip"):IsUnit()` now actually crashes the client rather than erroring with that usage message.

**Comment by Ketho** [comment]
> Additionally, `CreateFrame("GameTooltip"):IsUnit()` now actually crashes the client rather than erroring with that usage message.

Yeah, it seems to crash only on Release builds. It started crashing again when it went back to a Test build on 8.2.0 (30888)

**Comment by Meorawr** [comment]
[GetGuildBankMoneyTransaction](https://wow.gamepedia.com/API_GetGuildBankMoneyTransaction) is documented on the wiki as crashing the client if the given index is outside of the range 1 through `GetNumGuildBankMoneyTransactions()`, and is verifiable on my test character that isn't in a guild.

```lua
-- May require not being in a guild, or not having guild bank information loaded:
GetGuildBankMoneyTransaction(GetNumGuildBankMoneyTransactions());

-- These ones should always crash if the wiki is right (I've tested the first couple).
GetGuildBankMoneyTransaction(-1);
GetGuildBankMoneyTransaction(0);
GetGuildBankMoneyTransaction(GetNumGuildBankMoneyTransactions() + 1);
```

**Comment by Ketho** [comment]
As of patch 8.3.0 from these functions only `SetGuildRecruitmentComment()` and `GetGuildBankMoneyTransaction(-1)` can still crash the client

**Comment by Ketho** [comment]
Passing a negative languageId crashes the client, tested on 8.3.7 PTR and Live
```lua
SendChatMessage("test", nil, -1)
```
```lua
ERROR #128 (0x85100080) Attempt to access beyond the bounds of an array.

Program:	F:\World of Warcraft\_ptr_\WowT.exe
ProcessID:	3256
ThreadID:	15456
Object:	.?AV?$TSSimpleArray@H@@

index (0xFFFFFFFF), array size (0x0000011E)

<Exception.IssueType> Exception
<ExceptionType> Crash
<Exception.Summary:>
ERROR #128 (0x85100080) Attempt to access beyond the bounds of an array.
 index (0xFFFFFFFF), array size (0x0000011E)
.?AV?$TSSimpleArray@H@@
<:Exception.Summary>
```

**Comment by Meorawr** [comment]
The negative languageID crash is fixed in 9.0.1.35598 (maybe an earlier build, but time travel isn't my forté).
